homeassistant/components/mammotion/lawn_mower.py

```diff
@@ -0,0 +1,321 @@
+        # Return the specific attribute
+        return entity.attributes.get(attribute_name, None)
+    # Return None if the entity or attribute does not exist
+    return None
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: MammotionConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Luba config entry."""
+    mammotion_devices = entry.runtime_data
+
+    for mower in mammotion_devices:
+        async_add_entities([MammotionLawnMowerEntity(mower.reporting_coordinator)])
+
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_START_MOWING, START_MOW_SCHEMA, "async_start_mowing"
+    )
+
+    platform.async_register_entity_service(SERVICE_CANCEL_JOB, None, "async_cancel")
+
+
+class MammotionLawnMowerEntity(MammotionBaseEntity, LawnMowerEntity):
+    """Representation of a Mammotion lawn mower."""
+
+    _attr_supported_features = (
+        LawnMowerEntityFeature.DOCK
+        | LawnMowerEntityFeature.PAUSE
+        | LawnMowerEntityFeature.START_MOWING
+    )
+
+    def __init__(self, coordinator: MammotionReportUpdateCoordinator) -> None:
+        """Initialize the lawn mower."""
+        super().__init__(coordinator, "mower")
+        self._attr_name = None
+
+    @property
+    def rpt_dev_status(self) -> DeviceData:
+        """Return the device status."""
+        return self.coordinator.data.report_data.dev
+
+    @property
+    def report_data(self) -> ReportData:
+        """Return the report data."""
+        return self.coordinator.data.report_data
+
+    @property
+    def activity(self) -> LawnMowerActivity | None:
+        """Return the state of the mower."""
+
+        charge_state = self.rpt_dev_status.charge_state
+        mode = self.rpt_dev_status.sys_status
+        if mode is None:
+            return None
+
+        LOGGER.debug("activity mode %s", mode)
+        if mode == WorkMode.MODE_PAUSE or (
+            mode == WorkMode.MODE_READY and charge_state == 0
+        ):
+            return LawnMowerActivity.PAUSED
+        if mode == WorkMode.MODE_WORKING:
+            return LawnMowerActivity.MOWING
+        if mode == WorkMode.MODE_RETURNING:
+            return LawnMowerActivity.RETURNING
+        if mode == WorkMode.MODE_LOCK:
+            return LawnMowerActivity.ERROR
+        if mode == WorkMode.MODE_READY and charge_state != 0:
+            return LawnMowerActivity.DOCKED
+        return None
+
+    async def async_start_mowing(self, **kwargs: Any) -> None:
+        """Start mowing."""
+        trans_key = "pause_failed"
+
+        if kwargs:
+            await self.async_cancel()
+            entity_ids = kwargs.get("areas", [])
+
+            attributes = [
+                # TODO this should not need to be cast.
+                int(entity_hash)
+                for entity_id in entity_ids
+                if (entity_hash := get_entity_attribute(self.hass, entity_id, "hash"))
```

Comment:
What's this doing?

Reply:
Those are the area hashes, the value of an area to be mowed.

Reply:
Yes but why do we use a home assistant attribute for this?

Reply:
Yes but why do we use a home assistant attribute for this?

Reply:
Users sometimes need them for automations. Open to alternatives. The areas also have names and json data stored away, would it be more helpful if you knew the data structure?

Reply:
Well I'm more wondering why I have fetch date in her using an entity lookup instead of just from an internal source